src/python/pants/engine/internals/graph.py

```diff
@@ -323,11 +327,15 @@ async def resolve_target(
             return WrappedTarget(original_target)
     target = parametrizations.get(address)
     if target is None:
-        raise ValueError(
-            f"The address `{address}` was not generated by the target `{base_address}`, which "
-            "only generated these addresses:\n\n"
-            f"{bullet_list(str(t.address) for t in parametrizations.all)}\n\n"
-            "Did you mean to use one of those addresses?"
+        raise ResolveError(
+            softwrap(
+                f"""
+                The address `{address}` from {request.description_of_origin} was not generated by
+                the target `{base_address}`. It only generates these addresses:\n\n
+                {bullet_list(str(t.address) for t in parametrizations.all)}\n\n
+                Did you mean to use one of those addresses?
+                """
+            )
```

Comment:
This is the other way to formulate this, and this way is the one I prefer. More importantly though is to be consistent, either use "did you mean?" or "these are the known things: ...".

Oh, and now I noticed the "did you mean?" question at the end of the list. That is perfectly fine, and one of the issues I initially had in the first comment that the question was posed as "did you mean of these?" before presenting what the options were..

Reply:
Ah, and the reason is that I based my comments on what I read in the PR description, which now is out of date with the code... :P

Reply:
Based on your feedback, I made things consistent for both error messages. Thanks!